src/StatusQ/Controls/StatusBaseButton.qml

```diff
@@ -19,158 +22,109 @@ Rectangle {
-                opacity: !loading && statusBaseButton.icon.name !== ""
-                visible: statusBaseButton.icon.name !== ""
-                color: d.textColor
-            } // Icon
-            StatusEmoji {
-                width: statusBaseButton.icon.width
-                height: statusBaseButton.icon.height
-                anchors.verticalCenter: parent.verticalCenter
-                visible: statusBaseButton.icon.emoji
-                emojiId: Emoji.iconId(statusBaseButton.icon.emoji, statusBaseButton.icon.emojiSize) || ""
-            } // Emoji
-            StatusBaseText {
-                id: label
-                opacity: !loading
-                anchors.verticalCenter: parent.verticalCenter
-                font.pixelSize: size === StatusBaseButton.Size.Large ? 15 : 13 // by design
-
-                color: d.textColor
-            } // Text
-        } // Row
-
-
-        onPressed: {
-            if (!loading) {
-                statusBaseButton.pressed(mouse)
-            }
+    contentItem: RowLayout {
+        spacing: root.size === StatusBaseButton.Size.Large ? 6 : 4
+        StatusIcon {
+            Layout.preferredWidth: visible ? root.icon.width : 0
+            Layout.preferredHeight: visible ? root.icon.height : 0
+            icon: root.icon.name
+            rotation: root.asset.rotation
+            opacity: !loading && root.icon.name !== ""
+            visible: root.icon.name !== ""
+            color: d.textColor
         }
-
-        onReleased: {
-            if (!loading) {
-                statusBaseButton.released(mouse)
-            }
+        StatusEmoji {
+            Layout.preferredWidth: visible ? root.icon.width : 0
+            Layout.preferredHeight: visible ? root.icon.height : 0
+            visible: root.asset.emoji
+            emojiId: Emoji.iconId(root.asset.emoji, root.asset.emojiSize) || ""
         }
-
-        onClicked: {
-            if (!loading) {
-                statusBaseButton.clicked(mouse)
-            }
+        StatusBaseText {
+            Layout.alignment: Qt.AlignVCenter
+            opacity: !loading
+            font: root.font
+            text: root.text
+            color: d.textColor
+            verticalAlignment: Text.AlignVCenter
         }
+    }
 
-        onPressAndHold: {
-            if (!loading) {
-                statusBaseButton.pressAndHold(mouse)
-            }
+    Loader {
+        anchors.centerIn: parent
+        active: root.loading
+        sourceComponent: StatusLoadingIndicator {
+            color: d.textColor
         }
-    } // Sensor
+    }
+
+    // stop the mouse clicks in the "loading" state w/o disabling the whole button
+    // as this would make it impossible to have hover events or a tooltip
+    MouseArea {
+        anchors.fill: parent
+        acceptedButtons: Qt.AllButtons
+        enabled: root.loading
+        hoverEnabled: enabled
+        onPressed: mouse.accepted = true
+        onWheel: wheel.accepted = true
+    }
```

Comment:
Does this ensure that, when `loading`, there are no mouse related events emitted (clicked, pressed, etc) ?

Reply:
the `MouseArea` here is set to `enabled: root.loading`, @caybro shouldn't this be `enabled: !root.loading` and also shouldn't the `Button` be set to `enabled: !root.loading` as well?

Reply:
Pascal is right, this MouseArea is here to prevent the mouse items when the button is in the "loading" state; you can check using the sandbox app; no console msg should be printed when you click those

Reply:
Would love to check in sandbox, but can't run it anymore :D 

I trust you peoples that it works as intended!

Reply:
@caybro actually I put this mouse area there to get the pointing hand cursor, as there is no way currently to change the cursor shape of the QQC2 component, that's why it was inside the background property, however it can be also used to block events, but we should put the cursorShape back

Reply:
This can't work imho, esp. together with the fact we want the tooltip to appear eventhough we block the mouse events in the "loading" state